exercises/raindrops/src/Example.re

```diff
@@ -0,0 +1,24 @@
+let factor = (factor: int, input: int) : bool => input mod factor == 0;
+
+let makeSound = (t: (int, string), input: int) => {
+  let (n, sound) = t;
+  factor(n, input) ? Some(sound) : None;
+};
+
+let optionToString = (input: option(string)) : string =>
+  switch (input) {
+  | Some(string) => string
+  | None => ""
+  };
+
+let defaultIfEmpty = (input: string, default: string) : string =>
+  String.length(input) == 0 ? default : input;
+
```

Comment:
I tried to write the code

```
let [pling, plang, plong] = 
	[(3, "Pling"), (5, "Plang"), (7, "Plong")]
	|> List.map(makeSound)

```

and then use those `pling, plang, plong` here

```
let raindrops = (input: int) : string =>
  [pling, plang, plong]
  |> List.map(f => f(input))
  |> List.map(optionToString)
  |> String.concat("")
  |> defaultIfEmpty(_, string_of_int(input));
```

and it was working, but tests were failing, and also the compiler was complaining about I wasn't checking the case: `[]` on the `pling, plang, plong` destructuring. @stevejb71 do you have any idea why that was happening? Thank you in advance.

Reply:
Your latest code compiled when I cut & pasted it in - so I guess you got this sorted.

Reply:
@stevejb71 I did, but I don't understand why I was having that error 🤔

Reply:
Destructuring is like a pattern match, with only one case on the left hand side.

The compiler isn't smart enough to understand the three element case you gave is guaranteed to work, and gives you a warning that you've not handled the case of the right hand side being [].

Reply:
Thank you very much for your answer. I understand that compiler warning, but the thing is that tests were no passing. Maybe the compiler was aborting the compilation after the warning...

Reply:
If you have the code, I can give it a try locally.